apps/server/lib/lexical/server/project/progress/util.ex

```diff
@@ -0,0 +1,15 @@
+defmodule Lexical.Server.Project.Progress.Util do
+  alias Lexical.Server.Project.Progress
+
+  import Lexical.RemoteControl.Api.Messages
+
+  def with_progress(project, label, func) do
+    progress_pid = Progress.name(project)
+
+    send(progress_pid, project_progress(label: label <> ".begin"))
```

Comment:
I think you can get away from string concatenation and splitting if you just use a more descriptive record
```elixir
project_progress(label: label, stage: :begin)
```